nRF54H20DK: add support for nrfutil

gmarull opened the pull request.

Commit:
tests: sprintf: Avoid buffer overrun

fwrite parameters are "size_t size" and "size_t nmemb",
when writing a string we should set sizeof(char) and len(string).
The test is doing it wrongly and making the function read more
memory than it should.

Signed-off-by: Flavio Ceolin <flavio.ceolin@intel.com>

jonathonpenix committed:
cmake: llvm: riscv: Set RISC-V flags when building with clang

Ensure --target and -march/-mabi/-mcmodel are set appropriately when
building with clang targeting RISC-V.

Signed-off-by: Jonathon Penix <jpenix@quicinc.com>

jonathonpenix committed:
kconfig: llvm: riscv: Advertise TLS support when using llvm for RISC-V

clang and lld support TLS for RISC-V, so advertise this support in
Kconfig. I believe other non-RISC-V targets also have TLS support in LLVM,
but I'm not sure on the exact subset. As TLS support in LLVM wasn't
automatically advertised previously, gate this on RISC-V for now as it
a) shouldn't break other targets and b) prevents us from improperly
claiming support for unsupported targets.

Signed-off-by: Jonathon Penix <jpenix@quicinc.com>

moonlight83340 committed:
posix: Implement set and get scope APIs for pthread attr

Implement `pthread_attr_setscope()` and `pthread_attr_getscope()`
are required as part of _POSIX_THREAD_PRIORITY_SCHEDULING Option Group.

signed-off-by: Gaetan Perrot <gaetanperrotpro@gmail.com>

moonlight83340 committed:
doc: posix: mark set and get scope  APIs for pthread attr as supported

`pthread_attr_setscope()` and `pthread_attr_getscope()`
are now implemented, mark it so.

signed-off-by: Gaetan Perrot <gaetanperrotpro@gmail.com>

moonlight83340 committed:
doc: posix: mark putmsg as supported

Mark `putmsg()` as implemented but will fail with ``ENOSYS``
Was missing on the documentation before.

signed-off-by: Gaetan Perrot <gaetanperrotpro@gmail.com>

moonlight83340 committed:
posix: Implement tests for set and get scope APIs for pthread attr

Add tests for `pthread_attr_setscope()`
and `pthread_attr_getscope()`

signed-off-by: Gaetan Perrot <gaetanperrotpro@gmail.com>

Chenhongren committed:
drivers: timer: it8xxx2: enhances the accuracy of hw cycle calculation

This commit enhances the accuracy of hardware cycle calculation before
setting the IT8xxx2 event timer. The next target cycle is calculated by
the last, elapsed, and expected timeout ticks. And then, the difference
in hardware cycles between the target cycle and the current cycle is set
into the event timer. This increased accuracy effectively resolves the
clock drift issue.

Tested with:
west build -p always -b it8xxx2_evb tests/kernel/timer/timer_api
     -T kernel.timer.tickless
west build -p always -b it8xxx2_evb tests/kernel/timer/timer_behavior
     -T kernel.timer.timer

Fixes #67474 #67833

Signed-off-by: Ruibin Chang <Ruibin.Chang@ite.com.tw>
Signed-off-by: Ren Chen <Ren.Chen@ite.com.tw>

kv2019i committed:
arch: xtensa: add isync to interrupt vector

On Intel ADSP platforms, additional "isync" is needed in interrupt
vector to synchronize icache when core is woken up from deeper
sleep state by an interrupt. This is only needed if DSP clock
gating is enabled.

Signed-off-by: Kai Vehmanen <kai.vehmanen@linux.intel.com>

aescolar committed:
Bluetooth: Controller: Add coded phy radio timings for nrf52_bsim

Add Coded Phy radio timings for the nrf52_bsim

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>

aescolar committed:
nrf52_bsim: Enable CodedPhy support in DTS

The RADIO HW model supports coded phy now, let's enable it.

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>

danieldegrasse committed:
drivers: dma: dma_mcux_lpc: add workaround for RW61x flexspi

RW61x DMA has the *unique* restriction that DMA access is not routed
through the FlexSPI cache engine, only via the non-cached address space.
To enable DMA to read from the FlexSPI AHB space directly, fixup any
address passed to the DMA engine that is in the FlexSPI AHB cached
region to be in the non cached region

Signed-off-by: Daniel DeGrasse <daniel.degrasse@nxp.com>

pdgendt committed:
net: iface: Allow catch-all multicast monitor

Registering a multicast monitor with a NULL interface will receive all
events.

Signed-off-by: Pieter De Gendt <pieter.degendt@basalte.be>

pdgendt committed:
net: ethernet: mgmt: Add HW MAC address filtering

Add an ethernet management API to set or unset HW MAC address filtering.

Signed-off-by: Pieter De Gendt <pieter.degendt@basalte.be>

pdgendt committed:
net: ethernet: Register multicast monitor for MAC filtering

Register a multicast monitor for ethernet if HW MAC filtering is supported

Signed-off-by: Pieter De Gendt <pieter.degendt@basalte.be>

pdgendt committed:
drivers: ethernet: eth_mcux: Implement HW MAC address filtering

Replace the multicast monitor with a HW MAC filter configuration.

Signed-off-by: Pieter De Gendt <pieter.degendt@basalte.be>

pdgendt committed:
drivers: ethernet: eth_nxp_s32_gmac: Fix multicast filter

Update multicast filter with HW filtering and fix missing CONFIG_ prefix.

Signed-off-by: Pieter De Gendt <pieter.degendt@basalte.be>

pdgendt committed:
net: ip: Add helper NET_NATIVE_IP kconfig symbol

Introduce a NET_NATIVE_IP symbol similar to IPv4/IPv6 variants.

Signed-off-by: Pieter De Gendt <pieter.degendt@basalte.be>

pdgendt committed:
drivers: ethernet: eth_enet: Implement HW MAC address filtering

Replace the multicast monitor with a HW MAC filter configuration.

Signed-off-by: Pieter De Gendt <pieter.degendt@basalte.be>

erwango committed:
boards: nucleo_wba: Add flash partitions

Add flash partitions which can be used with mcuboot.

Signed-off-by: Erwan Gouriou <erwan.gouriou@st.com>

moonlight83340 committed:
doc: posix: sched functions will fail with ENOSYS

Mark `sched_setparam()` , `sched_setscheduler()`
and `sched_rr_get_interval()`
have implemented but will fail with ``ENOSYS``
Were missing on the documentation before.

signed-off-by: Gaetan Perrot <gaetanperrotpro@gmail.com>

pdgendt committed:
drivers: ethernet: eth_stm32: Rework HW MAC address filtering

Rework the driver to use HW MAC address filtering as an ethernet
capability.
Use a counter table for CRC indexes added/removed.

Signed-off-by: Pieter De Gendt <pieter.degendt@basalte.be>

pdgendt committed:
drivers: ethernet: eth_nxp_s32: Implement HW MAC address filtering

Replace the multicast monitor with a HW MAC filter configuration.

Signed-off-by: Pieter De Gendt <pieter.degendt@basalte.be>

TomChang19 committed:
drivers: espi: npcx: workaround Global Reset issue

Apply the workaround for the issue "eSPI global reset" in the
NPCX49nF_Errata

Signed-off-by: Tom Chang <CHChang19@nuvoton.com>

danieldegrasse committed:
treewide: fix board identifier in overlays for lpcxpresso55s69

Fix board identifier in overlays for the lpcxpresso55s69_cpu0 target.
Board identifiers were renamed with the following command:

for file in $(find . -name "lpcxpresso55s69_cpu0.*"); do;
	mv $file ${file/_cpu0/_lpc55s69_cpu0};
done

Additionally, remove lpcxpresso55s69_ns overlay targets where they are
no longer needed (or rename them where they are)

Signed-off-by: Daniel DeGrasse <daniel.degrasse@nxp.com>

danieldegrasse committed:
boards: update lpcxpresso55s69_cpu0 name in build commands

Update lpcxpresso55s69_cpu0 (and ns target) names in build commands in
the board documentation. Also, correct the documentation for dual core
images on this part, as it was out of date.

Signed-off-by: Daniel DeGrasse <daniel.degrasse@nxp.com>

danieldegrasse committed:
drivers: counter: counter_nxp_mrt: include soc.h for MRT CMSIS defines

Include soc.h in nxp MRT driver, so that CMSIS register definitions will
be available in this file

Signed-off-by: Daniel DeGrasse <daniel.degrasse@nxp.com>

henrikbrixandersen committed:
dts: bindings: can: make initial sample point properties optional

Make the properties for setting the initial sample points for both the
classic/arbitration phase and the data phase optional.

Signed-off-by: Henrik Brix Andersen <hebad@vestas.com>

henrikbrixandersen committed:
dts: bindings: can: remove optional sample point properties

Remove all optional, initial CAN sample point properties and rely on the
CAN timing calculations to automatically pick the preferred sample point
location based on the initial bitrate.

Signed-off-by: Henrik Brix Andersen <hebad@vestas.com>

henrikbrixandersen committed:
doc: releases: release-notes: 3.7: mention optional CAN DTS properties

Add a note about the ``sample-point`` and ``sample-point-data`` CAN
devicetree properties now being optional.

Signed-off-by: Henrik Brix Andersen <hebad@vestas.com>

aescolar committed:
Bluetooth: Controller: Nordic HAL: Use HAL to configure Timer's CC

Use the HAL always when setting the TIMERs CC registers,
(also for coded Phy) to ensure things work properly for simulation.

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>

hakehuang committed:
samples: tfm_integration: lpc55s69: enable tfm samples

Enable lpcxpresso55s69/lpc55s69/cpu0/ns for supported tfm samples

Signed-off-by: Hake Huang <hake.huang@oss.nxp.com>

marc-hb committed:
tests: llext: compile architectures not supported yet

For now llext supports a very limited number of architectures.  This
restriction is enforced by add_llext_target() in CMake at configuration
time.

Add a new `LOADER_BUILD_ONLY` conditional in tests/subsys/llext/simple/
and a new `llext.simple.loader_build`, `build_only` test that does not
invoke `add_llext_target()` and only compiles the llext framework code.

This helps find and fix bugs in `subsys/llext/*.c` and make it ready to
be used when add_llext_target() limitations are lifted.

Note this is pure `tests/` change without any change in the actual llext
framework code. The existing test is only modified to conditionally
invoke add_llext_target().

Signed-off-by: Marc Herbert <marc.herbert@intel.com>

anangl committed:
modules: hal_nordic: Align nRF54H20 nrfx_configs with nrfx templates

- add new entries that appeared in nrfx 3.4.0
- correct the default IRQ priority value for PPR (it is intended to be
  the lowest prority, so unlike for ARM cores, for RISC-V it should be
  the lowest value)

Signed-off-by: Andrzej Głąbek <andrzej.glabek@nordicsemi.no>

anangl committed:
dts: nordic: Add initial support for nRF54H20

Add definition of the nRF54H20 SoC with its Application, Radio,
and Peripheral Processor (PPR) cores and an initial set of
peripherals.

Signed-off-by: Andrzej Głąbek <andrzej.glabek@nordicsemi.no>

anangl committed:
soc: nordic: Add initial support for nRF54H20

Add Kconfig symbols that allow building for the nRF54H20 (not EngA) SoC.

Signed-off-by: Andrzej Głąbek <andrzej.glabek@nordicsemi.no>

anangl committed:
modules: hal_nordic: Add nrfx configuration needed for nRF54H20

Add cmake and nrfx_config entries that allow building for the nRF54H20
SoC.

Signed-off-by: Andrzej Głąbek <andrzej.glabek@nordicsemi.no>

mtpr-ot committed:
Bluetooth: controller: Add handling of HCI reset for sync_iso

Properly shut down ticker and initialize data at HCI reset.

Signed-off-by: Morten Priess <mtpr@oticon.com>

mtpr-ot committed:
Bluetooth: controller: Parameter check for ISO sync recv qualification

Add HCI input parameter validation according to spec for passing
qualification tests.

- Conditional compile of LE Set PA Receive Enable feature
- Store BIGinfo encryption state and number of streams
- Do not report BIGinfo if phy is invalid or unsupported
- Return disallowed ll_read_iso_tx_sync uses sync recever handle
- Validate BIS indices in ll_big_sync_create

Signed-off-by: Morten Priess <mtpr@oticon.com>

GTLin08 committed:
ITE: soc: it8xxx2: Add missing Kconfig file of it82302ax variant

Previous adjustments to hwmv2 lost this Kconfig file.

Test: west build -p always -b it82xx2_evb samples/hello_world
config BOARD_IT82XX2_EVB
   select SOC_IT82302_AX

Signed-off-by: Tim Lin <tim2.lin@ite.corp-partner.google.com>

erwango committed:
boards: st: nucleo_wba: Fix openocd configuration

Using Zephyr SDK 0.16.5, openocd can be used to debug WBA based boards.
Fix documentation and clean up board.cmake configuration for
nucleo_wba52cg from a local configuration which shouldn't have been
upstreamed

Signed-off-by: Erwan Gouriou <erwan.gouriou@st.com>

anangl committed:
boards: nordic: Add support for nRF54H20 DK

Add targets for the Application, Radio, and PPR cores in the nRF54H20
SoC on the nRF54H20 DK board.

Signed-off-by: Andrzej Głąbek <andrzej.glabek@nordicsemi.no>

anangl committed:
drivers: pinctrl_nrf: Set clockpin bitfield when necessary

For certain peripheral signal lines in nRF54H20, it is required
to set the clockpin bitfield for pins assigned to them, otherwise
the peripheral may not work properly, for example, there will be
no output from UART.

Signed-off-by: Andrzej Głąbek <andrzej.glabek@nordicsemi.no>

nashif committed:
doc: fix typo

Bluetooth -> Bluetoth

Fixes #67435

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
boards: xenvm: ignore userspace tests

Ignore userspace tag completely. Disabling USERSPACE in board kconfig is
not right, this will always be overidden by the application and we will
have build failures. The HW supports userspace and most filters in tests
depend on this capability.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

kartben committed:
boards: doc: Fix dead links in Arduino Nicla Sense ME doc page

Updated 3 dead (404) links in the doc page for Arduino Nicla Sense ME.
Also uniformized indenting of sphinx links in the process.

Signed-off-by: Benjamin Cabé <benjamin@zephyrproject.org>

kartben committed:
boards: Fix dead link in Laird Connectivity Sentrius BT610 Sensor page

Fix dead link to Sentrius™ BT610 Thermistor Coefficient Calculator page.

Signed-off-by: Benjamin Cabé <benjamin@zephyrproject.org>

kartben committed:
boards: doc: fix dead link to Keyestudio CAN-BUS Shield page

Fix an HTTP 404 for the link to Keyestudio CAN-BUS Shield product page.

Signed-off-by: Benjamin Cabé <benjamin@zephyrproject.org>

kartben committed:
doc: boards: Update E73-TBB product page link

E73-TBB board now lives at cdebyte.com.
Fixes #67099.

Signed-off-by: Benjamin Cabé <benjamin@zephyrproject.org>

kartben committed:
doc: boards: update links to DWM1001 web page

This board's docs are now to be found on Qorvo website, update links
accordingly.
Originally reported in #67099.

Signed-off-by: Benjamin Cabé <benjamin@zephyrproject.org>

kartben committed:
doc: boards: acn52832 remove 404 link

Aconno doesn't seem to be around anymore. Remove deadlink to product page.
Issue originally reported in #67099

Signed-off-by: Benjamin Cabé <benjamin@zephyrproject.org>

kartben committed:
doc: samples: use https URL for adt7420 sample

http://www.analog.com/adt7420 does not exist anymore nor redirect to
https, so update the link to be https

Signed-off-by: Benjamin Cabé <benjamin@zephyrproject.org>

kartben committed:
boards: intel_adsp: fix dead link

Chromium OS disk format documentation has moved.

Signed-off-by: Benjamin Cabé <benjamin@zephyrproject.org>

kartben committed:
asamples: drivers: ht16k33: fix product link

Update URL to product page to HTTPS & more recent version.

Signed-off-by: Benjamin Cabé <benjamin@zephyrproject.org>

aescolar committed:
Manifest: Update bsim to version v2.2

Which includes BLE coded phy support

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>

aescolar committed:
manifest: Update nRF hw models to latest

* Update the HW models module to
df94612a846b8b1da166d3e4e4273df54de02340

Including the following:
* df94612 GPIO: Add test API to be notified on output changes
* 0da9163 HW models: Use libUtil bs_create_folders_in_path
* c7798a4 CodedPhy: Several fixes and added documentation
* 534c8d4 RADIO: Move CCM decryption trigger to CRCEND
* bde13fc RADIO: Preliminary Coded Phy support
* ff2a79d RADIO: Refactored modulation selection
* f3823e5 RADIO: Fix in Tx packet end time
* d6ebdd9 RADIO: Refactored Address selection

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>

kartben committed:
doc: hwmv2: board_porting: fix broken links

DTS files for FRDM-K64F and Hexiwear K64 have moved.

Signed-off-by: Benjamin Cabé <benjamin@zephyrproject.org>

kartben committed:
boards: hwmv2: fix dead links for Laird Connect boards

typo in vendor name was causing dead links

Signed-off-by: Benjamin Cabé <benjamin@zephyrproject.org>

aescolar committed:
nrf53_bsim: Enable CodedPhy support in DTS

The RADIO HW model supports coded phy now, let's enable it.

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>

kartben committed:
boards: hwmv2: doc: fix dead links for Nordic

fixes a few broken links to Nordic DTS files

Signed-off-by: Benjamin Cabé <benjamin@zephyrproject.org>

Commit:
drivers: sensor: qdec_stm32: fix inverted polarity

Use the Polarity member instead of the ActivityInput.

Signed-off-by: Jeppe Odgaard <jeppe.odgaard@prevas.dk>

kartben committed:
doc: boards: hwmv2: fix dead link for rpi4b

fix typo in raspberrypi vendor name causing dead link

Signed-off-by: Benjamin Cabé <benjamin@zephyrproject.org>

kartben committed:
doc: boards: hwmv2: fix dead links for Seeed

Seeed Studio boards live in boards/seeed so update file links accordingly.

Signed-off-by: Benjamin Cabé <benjamin@zephyrproject.org>

kartben committed:
doc: boards: silabs: fix bad links

fix minor typos causing dead links

Signed-off-by: Benjamin Cabé <benjamin@zephyrproject.org>

kartben committed:
doc: boards: hwmv2: fix dead links to u-blox DTS fils

U-Blox boards live in /boards/u-blox, not /boards/ublox
This commit fixes bad links accordingly.

Signed-off-by: Benjamin Cabé <benjamin@zephyrproject.org>

kartben committed:
doc: boards: hwmv2: fix dead links to Wurth Electronik DTS fils

Wurth Electronik boards live in /boards/we, not /boards/wurth_elektronik
This commit fixes bad links accordingly.

Signed-off-by: Benjamin Cabé <benjamin@zephyrproject.org>

kartben committed:
doc: soc: hwmv2: fix altera dead links

fixing dead links to some Altera SoC files

Signed-off-by: Benjamin Cabé <benjamin@zephyrproject.org>

kartben committed:
doc: fix double backticks in sphinx roles

fixed a few improperly formatted sphinx roles causing dead references.

Signed-off-by: Benjamin Cabé <benjamin@zephyrproject.org>

rlubos committed:
net: tcp: Remove redundant debug logs

Debug logs in helper functions like tcp_unsent_len() or
tcp_window_full() are not very helpful and generate a heavy, unnecessary
log output. Therefore, tcp_unsent_len() will no longer generate log, and
tcp_window_full() will print out a log only when the window is actually
full, which could be an useful information.

Also, reduce the log load during TX, as currently redundant logs were
printed in tcp_out_ext(), tcp_send_process_no_lock() and finally in
tcp_send().

Signed-off-by: Robert Lubos <robert.lubos@nordicsemi.no>

aescolar committed:
ci: Ensure twister steps use the bsim version from the manifest

The west manifest may point to a newer version of bsim than
the one avaliable in the docker image. Let's ensure
we run with that one.
This is equivalent to the change done in
933d338c9783cce55466fed000baf2f3beeac891
for the bsim-tests workflow.

Note that the update and rebuild is incremental,
so if the docker image already has the right
version the operation is very fast
(~1 second)

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>

kartben committed:
doc: boards: hwmv2: fix dead links to Broadcom files

Broadcom boards live in /boards/brcm, not /boards/broadcom
This commit fixes bad links accordingly.

Signed-off-by: Benjamin Cabé <benjamin@zephyrproject.org>

kartben committed:
doc: hwmv2: boards: fix broken NXP links

fix bad links to some NXP board files (and make references actual "links")

Signed-off-by: Benjamin Cabé <benjamin@zephyrproject.org>

kartben committed:
doc: guidelines: fix dead links

fix zephyr_file and zephyr_raw broken links

Signed-off-by: Benjamin Cabé <benjamin@zephyrproject.org>

kartben committed:
doc: boards: hwmv2: fix teensy dead link

fix a dead reference to teensy40_defconfig file

Signed-off-by: Benjamin Cabé <benjamin@zephyrproject.org>

kartben committed:
doc: boards: hwmv2: fix seagate faze dead link

fix a dead reference to faze/support/openocd.cfg

Signed-off-by: Benjamin Cabé <benjamin@zephyrproject.org>

kartben committed:
doc: dts: fix link to reel board DTS file

fix broken link following hwmv2 migration

Signed-off-by: Benjamin Cabé <benjamin@zephyrproject.org>

anangl committed:
samples: Replace nRF54H20 PDK with nRF54H20 DK

nRF54H20 PDK is superseded by nRF54H20 DK and will no longer be
supported.

Signed-off-by: Andrzej Głąbek <andrzej.glabek@nordicsemi.no>

anangl committed:
snippets: Replace nRF54H20 PDK with nRF54H20 DK

nRF54H20 PDK is superseded by nRF54H20 DK and will no longer be
supported.

Signed-off-by: Andrzej Głąbek <andrzej.glabek@nordicsemi.no>

anangl committed:
tests: Replace nRF54H20 PDK with nRF54H20 DK

nRF54H20 PDK is superseded by nRF54H20 DK and will no longer be
supported.

Signed-off-by: Andrzej Głąbek <andrzej.glabek@nordicsemi.no>

anangl committed:
boards: nordic: Remove definition of nRF54H20 PDK

nRF54H20 PDK is superseded by nRF54H20 DK and will no longer be
supported.

The board was superseded by nRF54H20 DK.

Signed-off-by: Andrzej Głąbek <andrzej.glabek@nordicsemi.no>

anangl committed:
modules: hal_nordic: Remove support for nRF54H20 EngA

This was a preview revision of the SoC that will no longer
be supported.

Signed-off-by: Andrzej Głąbek <andrzej.glabek@nordicsemi.no>

anangl committed:
dts: Remove support for nRF54H20 EngA

This was a preview revision of the SoC that will no longer
be supported.

Signed-off-by: Andrzej Głąbek <andrzej.glabek@nordicsemi.no>

anangl committed:
soc: nordic: Remove support for nRF54H20 EngA

This was a preview revision of the SoC that will no longer
be supported.

Signed-off-by: Andrzej Głąbek <andrzej.glabek@nordicsemi.no>

kv2019i committed:
soc: intel_adsp_cavs: store PS when power gating secondary core

When non-primary core is powered down and restart with sequence of:
 - PM state set to SOFT_OFF
 - once target core is idle, cut power with soc_adsp_halt_cpu()
 - power up core again with k_smp_cpu_resume()

The execution will continue from stored DSP core context, but
will hit an assert in z_smp_cpu_mobile() as the PS.INTLEVEL
is zero.

Fix this issue by storing and restoring PS register in this flow.

Link: https://github.com/zephyrproject-rtos/zephyr/issues/70181
Signed-off-by: Kai Vehmanen <kai.vehmanen@linux.intel.com>

SamyFrancelet committed:
drivers: eswifi: Add option for maximum data size

Default MAX_DATA_SIZE might be too small depending
on application (e.g. handling the scan result with 12 APs detected)
Added config option for MAX_DATA_SIZE and warning for the user

Signed-off-by: Samy Francelet <samy.francelet@ik.me>

Commit:
soc: nxp: rw: Clock SPI Flexcomms

Clock flexcomms if used as SPI

Signed-off-by: Declan Snyder <declan.snyder@nxp.com>

Commit:
boards: rd_rw612_bga: Enable FC0 as SPI

Enable flexcomm0 as SPI on rd_rw612_bga

Signed-off-by: Declan Snyder <declan.snyder@nxp.com>

Commit:
tests: spi_loopback: Support rd_rw612_bga

Add rd_rw612_bga to spi loopback test

Signed-off-by: Declan Snyder <declan.snyder@nxp.com>

danieldegrasse committed:
dts: arm: nxp: rw6xx: add dma0 definition

Add DMA0 definition for RW61x SOC

Signed-off-by: Daniel DeGrasse <daniel.degrasse@nxp.com>

danieldegrasse committed:
boards: nxp: rd_rw612_bga: enable DMA support

Enable support for dma controller on RD RW612 BGA board, and add overlay
to enable board in dma loop transfer test

Signed-off-by: Daniel DeGrasse <daniel.degrasse@nxp.com>

aescolar committed:
ci: bsim update step: Do not affect global git config

When updating the bsim install in the docker image's
/opt/bsim_west to the version pointed in the Zephyr
manifest, we are detaching the head to that version.
We suppress the "You are in 'detached HEAD' state..."
git warning as it is just noise in CI.

So far we were doing it by globally supressing that
warning. Let's instead just supress it for the
command in question so we don't change global
git config against expectations.

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>

57300 committed:
snippets: Remove nordic-ppr-ram

The `nordic-ppr` snippet can now be used instead, since RAM execution is
default for PPR.

Signed-off-by: Grzegorz Swiderski <grzegorz.swiderski@nordicsemi.no>

57300 committed:
boards: nrf54h20dk: Make execution from RAM default for PPR

In practice, PPR is intended to be running code from RAM, so make this
the default choice for the `nrf54h20dk/nrf54h20/cpuppr` board target.
Keep the MRAM execution option as a `xip` variant of that target,
replacing the `ram` one.

Align the default `cpuapp` configuration for copying PPR's image to RAM
before it boots the child processor.

Signed-off-by: Grzegorz Swiderski <grzegorz.swiderski@nordicsemi.no>

57300 committed:
snippets: Add nordic-ppr-xip

This replacement for the `nordic-ppr-ram` snippet does the opposite:
enable PPR execution in place from MRAM.

Signed-off-by: Grzegorz Swiderski <grzegorz.swiderski@nordicsemi.no>

57300 committed:
drivers: misc: nordic_vpr_launcher: Update src/exec memory size check

Firstly, build-asserting the execution/source memory sizes to be equal
wasn't working, due to the wrong (non-inst) DT API being used.

Secondly, this assert can be relaxed so that the source memory region
only needs to have greater than or equal size to the execution region,
as VPR firmware needs to fit into execution memory first and foremost.
This will come in handy, since MRAM partitions (typical source memory)
have stricter alignment requirements than RAM regions.

Signed-off-by: Grzegorz Swiderski <grzegorz.swiderski@nordicsemi.no>

JiafeiPan committed:
tests: gpio_basic_api: change imx93 evk board name

Updated according to hardware model v2 board name changes.

Signed-off-by: Jiafei Pan <Jiafei.Pan@nxp.com>

nordicjm committed:
sysbuild: Add support for COMPILER_WARNINGS_AS_ERRORS

Adds support for the Kconfig option to enable compiler warnings
being treated as errors in all images that are build. This is a
sticky-set option which means enabling it in sysbuild will
enable it in all images, if it is then disabled in sysbuild it
will not be disabled in any of the images as a result and would
need to be manually unselected.

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

PerMac committed:
twister: Use proper warnings-to-errors command for sysbuild

If sysbuild is used a flag SB_CONFIG_COMPILER_WARNINGS_AS_ERRORS
has to be used in order to turn warings to errors on all images.
Align pytest tests with the change.

Fixes: #67360

Signed-off-by: Maciej Perkowski <Maciej.Perkowski@nordicsemi.no>

aescolar committed:
native_sim: Add twister definition for native_sim/native/64

Which will replace the old native_sim_64 to align it with
the hwmv2 naming conventions.

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>

aescolar committed:
native_posix: Add twister definition for native_sim/native/64

Which will replace the old native_sim_64 to align it with
the hwmv2 naming conventions.

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>

PavelVPV committed:
Bluetooth: Mesh: Warn if can't listen to unprov beacons until proved

When calling `mesh prov beacon-listen on` shell command before
provisioning the local device, the stack won't report unprovisioned
beacons. This is because the scanner has not been started yet. This
behavior confuses users. Print a warning that the device needs to be
provisioned first before it can listen to unprovisioned beacons.

Signed-off-by: Pavel Vasilyev <pavel.vasilyev@nordicsemi.no>

aescolar committed:
samples/net/telnet: Filter out for NATIVE_APPLICATION

This sample enables EVENTFD (through NET_SOCKETS_SERVICE)
which is not kconfig compatible with the old
NATIVE_APPLICATION targets (native_posix[//64])
Let's explicity filter out this platforms to avoid the
sample from failing to build

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>

aescolar committed:
docs/Kconfig help: Replace native_[posix,sim]_64

Replace uses of native_[posix,sim]_64 in the documentation
and kconfig help messages with native_[posix,sim]/native/64
to match the hwmv2 naming.

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>

aescolar committed:
arch/posix cmake: Replace native_posix w native_sim and use hwmv2 names

Let's fix the error messages:
* native_posix is in general replaced with native_sim
* let's use hwmv2 names

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>

aescolar committed:
samples lvgl/accelerometer_chart: Use hwmv2 native targets

For the 64 bit targets, change identifiers to the new hwmv2 ones.
And remove redundant overlays (which were equal for the 32
and 64 bit versions)

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>

aescolar committed:
samples lvgl/demos: Use hwmv2 native targets

For the 64 bit targets, change identifiers to the new hwmv2 ones.
And remove redundant overlays (which were equal for the 32
and 64 bit versions)

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>

aescolar committed:
samples/drivers: Use hwmv2 native targets

For the 64 bit targets, change identifiers to the new hwmv2 ones.

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>

aescolar committed:
samples/net/gptp: Use hwmv2 native targets identifiers

For the 64 bit targets, change identifiers to the new hwmv2 ones.
And remove redundant overlays (which were equal for the 32
and 64 bit versions)

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>

aescolar committed:
samples/net/gptp: Use hwmv2 native targets identifiers

For the 64 bit targets, change identifiers to the new hwmv2 ones.
And remove redundant overlays (which were equal for the 32
and 64 bit versions)

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>

aescolar committed:
samples/net/sockets/*: Use hwmv2 native targets identifiers

For the 64 bit targets, change identifiers to the new hwmv2 ones.
And remove redundant overlays (which were equal for the 32
and 64 bit versions)

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>

aescolar committed:
samples/subsys/display/lvgl: Use hwmv2 native targets identifiers

For the 64 bit targets, change identifiers to the new hwmv2 ones.
And remove redundant overlays (which were equal for the 32
and 64 bit versions)

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>

aescolar committed:
samples/subsys/settings: Use hwmv2 native targets identifiers

For the 64 bit targets, change identifiers to the new hwmv2 ones.
And remove redundant overlays (which were equal for the 32
and 64 bit versions)

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>

aescolar committed:
samples/subsys/zbus/uart_bridge: Remove redundant overlays

Remove redundant overlays (which were equal for the 32
and 64 bit versions)

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>

aescolar committed:
samples/net several: Use hwmv2 native targets identifiers

For the 64 bit targets, change identifiers to the new hwmv2 ones.
And remove redundant overlays (which were equal for the 32
and 64 bit versions)

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>

aescolar committed:
samples/posix: Use hwmv2 native targets identifiers

For the 64 bit targets, change identifiers to the new hwmv2 ones.
And remove redundant overlays (which were equal for the 32
and 64 bit versions)

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>

aescolar committed:
samples/subsys: Use hwmv2 native targets identifiers

For the 64 bit targets, change identifiers to the new hwmv2 ones.

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>

aescolar committed:
samples misc: Use hwmv2 native targets identifiers

For the 64 bit targets, change identifiers to the new hwmv2 ones.

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>

aescolar committed:
tests/bluetooth: Use hwmv2 native targets identifiers

For the 64 bit targets, change identifiers to the new hwmv2 ones.
And remove redundant overlays (which were equal for the 32
and 64 bit versions)

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>

aescolar committed:
tests/drivers/adc: Use hwmv2 native targets identifiers

For the 64 bit targets, change identifiers to the new hwmv2 ones.
And remove redundant overlays (which were equal for the 32
and 64 bit versions)

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>

aescolar committed:
tests/drivers/build_all: Use hwmv2 native targets identifiers

For the 64 bit targets, change identifiers to the new hwmv2 ones.
And remove redundant overlays (which were equal for the 32
and 64 bit versions)

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>

aescolar committed:
tests/drivers/dma: Use hwmv2 native targets identifiers

For the 64 bit targets, change identifiers to the new hwmv2 ones.
And remove redundant overlays (which were equal for the 32
and 64 bit versions)

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>

aescolar committed:
tests/subsys/logging: Use hwmv2 native targets identifiers

For the 64 bit targets, change identifiers to the new hwmv2 ones.
And remove redundant overlays (which were equal for the 32
and 64 bit versions)

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>

aescolar committed:
tests/drivers/gpio: Use hwmv2 native targets identifiers

For the 64 bit targets, change identifiers to the new hwmv2 ones.
And remove redundant overlays (which were equal for the 32
and 64 bit versions)

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>

aescolar committed:
tests/drivers/input: Use hwmv2 native targets identifiers

For the 64 bit targets, change identifiers to the new hwmv2 ones.
And remove redundant overlays (which were equal for the 32
and 64 bit versions)

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>

aescolar committed:
tests/drivers/flash_simulator: Use hwmv2 native targets identifiers

For the 64 bit targets, change identifiers to the new hwmv2 ones.
And remove redundant overlays (which were equal for the 32
and 64 bit versions)

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>

aescolar committed:
tests/drivers/udc: Use hwmv2 native targets identifiers

For the 64 bit targets, change identifiers to the new hwmv2 ones.

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>

aescolar committed:
tests/drivers/kscan: Use hwmv2 native targets identifiers

For the 64 bit targets, change identifiers to the new hwmv2 ones.
And remove redundant overlays (which were equal for the 32
and 64 bit versions)

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>

aescolar committed:
tests/drivers several: Use hwmv2 native targets identifiers

For the 64 bit targets, change identifiers to the new hwmv2 ones.

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>

aescolar committed:
tests/net: Use hwmv2 native targets identifiers

For the 64 bit targets, change identifiers to the new hwmv2 ones

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>

aescolar committed:
tests/boards/native_sim: Use hwmv2 native targets identifiers

For the 64 bit targets, change identifiers to the new hwmv2 ones

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>

aescolar committed:
tests/lib: Use hwmv2 native targets identifiers

For the 64 bit targets, change identifiers to the new hwmv2 ones

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>

aescolar committed:
tests/misc/check_init_priorities: Use hwmv2 native targets ident

For the 64 bit targets, change identifiers to the new hwmv2 ones
And remove redundant overlays (which were equal for the 32
and 64 bit versions)

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>

aescolar committed:
tests/posix: Use hwmv2 native targets identifiers

For the 64 bit targets, change identifiers to the new hwmv2 ones

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>

aescolar committed:
tests/subsys/settings: Use hwmv2 native targets identifiers

For the 64 bit targets, change identifiers to the new hwmv2 ones.
And remove redundant overlays (which were equal for the 32
and 64 bit versions)

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>

aescolar committed:
tests/subsys/input: Use hwmv2 native targets identifiers

For the 64 bit targets, change identifiers to the new hwmv2 ones.
And remove redundant overlays (which were equal for the 32
and 64 bit versions)

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>

aescolar committed:
tests/subsys/fs: Use hwmv2 native targets identifiers

For the 64 bit targets, change identifiers to the new hwmv2 ones.
And remove redundant overlays (which were equal for the 32
and 64 bit versions)

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>

aescolar committed:
tests/subsys/dfu: Use hwmv2 native targets identifiers

For the 64 bit targets, change identifiers to the new hwmv2 ones.
And remove redundant overlays (which were equal for the 32
and 64 bit versions)

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>

aescolar committed:
tests/subsys several: Use hwmv2 native targets identifiers

For the 64 bit targets, change identifiers to the new hwmv2 ones.

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>

aescolar committed:
tests/ztest: Use hwmv2 native targets identifiers

For the 64 bit targets, change identifiers to the new hwmv2 ones

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>

aescolar committed:
tests/modules/nanopb: Use hwmv2 native targets identifiers

For the 64 bit targets, change identifiers to the new hwmv2 ones

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>

aescolar committed:
native_posix: Remove twister definition for old native_sim/posix_64

Remove the hwmv1 twister board definitions for
native_posix_64 / native_sim_64 in preparation to
transition to the new definitions for hwmv2

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>

mmahadevan108 committed:
boards: frdm_mcxn947: Move the GPIO button nodes to the common file

Move the GPIO button dts nodes to the common dts file so it
be selected by either cpu core.
Also fix an error in the GPIO setting for one of the switches.

Signed-off-by: Mahesh Mahadevan <mahesh.mahadevan@nxp.com>

Thalley committed:
Bluetooth: CAP: Fix issue with codec_cfg in CAP uni start

The codecs used for the bap bt_bap_stream_config was only
valid for the lifetime of the procedure, which meant
that by the end of the procedure the stream->codec_cfg
became invalid.

This is fixed by using the pointer provided to the
CAP API, and documentating the lifetime of the codec_cfg.

Signed-off-by: Emil Gydesen <emil.gydesen@nordicsemi.no>

Thalley committed:
Bluetooth: TMAP: Shell: Improve logging of role

The role is a 16-bit value, and is best logged as %04X.

Signed-off-by: Emil Gydesen <emil.gydesen@nordicsemi.no>

fengming-ye committed:
net: ipv6: fix link local ping fail with default router

When we receive Router Advertisement with life time,
we will add this as default router, like typing command "net iface"
and has show below
IPv6 default router :
fe80:xxxx::xxxx

When this default router is backend A and
we ping backend B with link local address,
we will use default router A and send echo request
to backend A instead of B, which will receive Redirect and no reply.

Fix it by link local address does not check route.

Signed-off-by: Fengming Ye <frank.ye@nxp.com>

JiafeiPan committed:
boards: imx93_evk: add led1 alias

So that it can be used to run basic thread sample application.

Signed-off-by: Jiafei Pan <Jiafei.Pan@nxp.com>

nordicjm committed:
samples: ipc: rpmsg_service: Add sample file, remove broken builds

Adds a sample file so that this sample is built in CI and removes
all configurations that do not build

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
boards: lairdconnect: bl5340_dvk: Fix file name

Fixes an wrong file name for a file that had been renamed as part
of the hwmv2 port

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

ad8sweera committed:
net: ipv4: Allow IGMP packets with zero src address

IGMP queries sent out by a proxy querier can have
a source IP address 0.0.0.0
Allow these incoming packets.

Fixes #69917

Signed-off-by: Savin Weeraratne <savin.weeraratne@audinate.com>

aescolar committed:
Bluetooth: Controller: Correct PPI->Timer start delay for simulation

In the simulated bsim boards there is no jittery delay
from a (D)PPI event until a TIMER starts.
The timer starts instantaneously so we do not need to
compensate for it.

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>

aescolar committed:
tests/drivers/build_all/w1: Run on native_sim instead of native_posix

native_posix is being replaced with native_sim, let's
have these tests be run on native_sim instead.

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>

aescolar committed:
samples/basic/hash_map: Remove clarification sentence

This current sentence would need to be clarified a bit
further to be really correct:
It is only when building with the host C library that it
applies, but with native_sim we can build with other
libC's.
The twister tests definitions are testing only with
embedded C libraries,
and there is anyway no harm if somebody defines the heap
size even if using the host libC.
So as such it would seem better to just remove this
sentence.

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>

ndrs-pst committed:
drivers: mdio: fix incorrect return usage in `mdio_bus_enable/disable`

Remove incorrect return statements in the `z_impl_mdio_bus_enable` and
`z_impl_mdio_bus_disable` functions within the MDIO driver API.
These functions are intended to call the `bus_enable` and `bus_disable`
methods of the MDIO driver API without returning a value,
as they are defined to return void.

Signed-off-by: Pisit Sawangvonganan <pisit@ndrsolution.com>

aescolar committed:
tests/drivers/build_all/w1: Add gpio dependency

This driver enables CONFIG_GPIO, so if we try
to target a board which does not support it it
will fail.
Let's add the required tag in the testcase yaml.

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>

lylezhu2012 committed:
Bluetooth: Host: Change __line__ to __LINE__

The building error is reported in Ubuntu if
CONFIG_NET_BUF_LOG is true.

Change __line__ to __LINE__ to fix building
error.

Signed-off-by: Lyle Zhu <lyle.zhu@nxp.com>

gmarull committed:
scripts: west_commands: runners: nrfjprog: drop nRF54H support

nRF54Hx series is no longer supported in nrfjprog. nrfutil is now
required.

Signed-off-by: Gerard Marull-Paretas <gerard@teslabs.com>

gmarull committed:
scripts: west_commands: runners: nrfutil: use x-execute-batch

execute-batch is now named x-execute-batch, as it is an experimental
option.

Signed-off-by: Gerard Marull-Paretas <gerard@teslabs.com>

gmarull committed:
scripts: west_commands: runners: nrfutil: provide live feedback

nrfutil runner uses the batch-mode, so no live feedback is provided to
the user. However, batch-mode reports batch progress containing
human-readable strings of the operation being done. This patch changes
the _exec() implementation to parse the subprocess output in real-time,
logging to info the 'batch_update' reports. Note that only the first
batch update of a sequence (percentage = 0) is logged because first,
percentage resolution seems to be pretty bad, and, because logging
messages cannot be easily _appended_.

Signed-off-by: Gerard Marull-Paretas <gerard@teslabs.com>

gmarull committed:
scripts: west_commands: runners: nrf_common: update nRF54H support

nRF54H can only be flashed using nrfutil now, so some workaround present
in the nrf_common module are no longer needed, e.g. UICR erasing.

Signed-off-by: Gerard Marull-Paretas <gerard@teslabs.com>

gmarull committed:
boards: nordic: nrf54h20dk: enable nrfutil

Board can be programmed using nrfutil.

Signed-off-by: Gerard Marull-Paretas <gerard@teslabs.com>

nashif committed:
soc: esp32: make SCHED_CPU_MASK depend on SCHED_DUMB

Forcing SCHED_CPU_MASK without SCHED_DUMB results in a global warning
from Kconfig.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

carlescufi commented on scripts/west_commands/runners/nrfutil.py:
```diff
@@ -40,27 +40,31 @@ def do_create(cls, cfg, args):
-        return out
+        jout_all = []
+        cmd = ['nrfutil', '--json', 'device'] + args
+        with subprocess.Popen(cmd, stdout=subprocess.PIPE) as p:
```
Please log the command and allow for dry runs like here:
https://github.com/zephyrproject-rtos/zephyr/blob/99f62a749cbbb0da676b6320c2c9492e3914b2d1/scripts/west_commands/runners/core.py#L746-L748

carlescufi commented on scripts/west_commands/runners/nrfutil.py:
```diff
@@ -40,27 +40,31 @@ def do_create(cls, cfg, args):
-        else:
-            cpe = None
-        finally:
-            # https://github.com/ndjson/ndjson-spec
```
Please include this link, it is necessary to understand the format.

carlescufi commented on scripts/west_commands/runners/nrfutil.py:
```diff
@@ -40,27 +40,31 @@ def do_create(cls, cfg, args):
+        jout_all = []
+        cmd = ['nrfutil', '--json', 'device'] + args
+        with subprocess.Popen(cmd, stdout=subprocess.PIPE) as p:
+            while True:
```
why not `while p.stdout.readline()` ?

gmarull replied:
this doesn't store the line output?

gmarull replied:
pls check latest revision

nordic-piks commented on scripts/west_commands/runners/nrf_common.py:
```diff
@@ -92,11 +86,6 @@ def do_add_parser(cls, parser):
                             help='''erase all user available non-volatile
                             memory and disable read back protection before
                             flashing (erases flash for both cores on nRF53)''')
-        parser.add_argument('--erase-all-uicrs', required=False,
```
What would be the procedure to erase unused core?
I.e.. There was fw written at APP core (empty RAD), then I want to run application at Radio core.
How can I prevent execution at APP core?
Previously this was done by this arg = --erase-all-uicrs

gmarull replied:
now `west flash --erase` should just work (see latest revision)

carlescufi reviewed (approved):
LGTM now, thanks!

carlescufi commented on scripts/west_commands/runners/nrfutil.py:
```diff
@@ -40,27 +42,34 @@ def do_create(cls, cfg, args):
+            return {}
+
+        with subprocess.Popen(cmd, stdout=subprocess.PIPE) as p:
+            for line in iter(p.stdout.readline, b''):
```
Actually, this would be simpler with:

`for line in p.stdout:`

carlescufi replied:
Although this might not behave the same way when it comes to buffering the stdout, so feel free to ignore this suggestion.

aescolar committed:
tests/bsim: Reorder tests to avoid failure in GATT/SR/GAS/BV-01-C

GATT/SR/GAS/BV-01-C has just broken due to what appears
like a minor timing change.
This indicates the test may be too time sensitive.
Just reordering the tests, it passes again.
So let's reorder the test list, and we get the minor
benefit of havign them move alphabetically ordered.

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>

maass-hamburg committed:
mgmt: hawkbit: Fix condition in hawkbit_probe

Fix send_request condition in hawkbit_probe function

Signed-off-by: Fin Maaß <f.maass@vogl-electronic.com>
